AppCenterDistribute/AppCenterDistribute/MSDistribute.m

```diff
@@ -807,6 +809,31 @@ - (NSMutableDictionary *)getReportingParametersForUpdatedRelease:(NSString *)upd
   return reportingParameters;
 }
 
+- (void)changeDistributionGroupIdAfterAppUpdateIfNeeded:(NSString *)currentInstalledReleaseHash {
+  NSString *lastDownloadedDistributionGroupId = [MS_USER_DEFAULTS objectForKey:kMSDownloadedDistributionGroupIdKey];
+  if (lastDownloadedDistributionGroupId == nil) {
+    return;
+  }
+
+  // Skip if the current release was not updated.
+  NSString *lastDownloadedReleaseHashes = [MS_USER_DEFAULTS objectForKey:kMSDownloadedReleaseHashKey];
+  if ((lastDownloadedReleaseHashes == nil) || ([lastDownloadedReleaseHashes rangeOfString:currentInstalledReleaseHash].location == NSNotFound)) {
+    return;
+  }
+
+  // Skip if the group ID of an updated release is the same as the current one.
+  NSString *currentDistributionGroupId = [MS_USER_DEFAULTS objectForKey:kMSDistributionGroupIdKey];
+  if ((currentDistributionGroupId == nil) || ([lastDownloadedDistributionGroupId isEqualToString:currentDistributionGroupId] == NO)) {
+
+    // Set group ID from downloaded release details if an updated release was downloaded from another distribution group.
+    MSLogDebug([MSDistribute logTag], @"Current group ID doesn't match the group ID of downloaded release, updating current group id: %@", lastDownloadedDistributionGroupId);
+    [MS_USER_DEFAULTS setObject:lastDownloadedDistributionGroupId forKey:kMSDistributionGroupIdKey];
```

Comment:
What I mentioned is that it seems `currentDistributionGroupId` is the new value and `lastDownloadedDistributionGroupId` is the latest value stored in `NSUserDefaults`.
The value doesn't match which means it needs to be updated but you are storing the old value, not the new one.
Please correct me if I'm wrong.

Reply:
`currentDistributionGroupId` is used to store the group ID of currently running release (before the update). `lastDownloadedDistributionGroupId` to store the ID of a release that was downloaded before app update and restart. So the workflow is:
- In in-ap update dialog a user press "Download". SDK saves details about the downloaded release (`lastDownloadedDistributionGroupId`).
- After an app restart, SDK checks if currently running release was actually updated from downloaded IPA.
- If an updated release was downloaded from another distribution group then we need to update the stored ID (move the user to another group).

I updated the variables names so I hope it's more clear now. Thank you.